### PR TITLE
build: update protobuf-build to support apple silicon

### PR DESCRIPTION
**Pull request by xie-zheng**
I hereby agree to the terms of the [GreptimeDB CLA](https://gist.github.com/xtang/6378857777706e568c1949c7578592cc)

## What's changed and what's your intention?

So, the story is that I want to build the greptimedb on my computer(Apple M1 & macOS 13.4.1 & libprotoc 23.4),
and I got an Error!

```
  --- stderr
  thread 'main' panicked at 'No suitable `protoc` (>= 3.1.0) found in PATH', 
  /Users/xiezheng/.cargo/registry/src/rsproxy.cn-0dccff568467c15b/protobuf-build-0.14.1/src/protobuf_impl.rs:35:14
  note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

I go checked the source code in protobuf-build(v0.14.1)

```rust
    // The bundled protoc should always match the version
    let protoc_bin_name = match (env::consts::OS, env::consts::ARCH) {
        ("linux", "x86") => "protoc-linux-x86_32",
        ("linux", "x86_64") => "protoc-linux-x86_64",
        ("linux", "aarch64") => "protoc-linux-aarch_64",
        ("linux", "powerpc64") => "protoc-linux-ppcle_64",
        ("macos", "x86_64") => "protoc-osx-x86_64",
        ("windows", _) => "protoc-win32.exe",
        
         // no considering for m1 mac!!!     
    
        /* ! ==> PANIC HERE <== */
        _ => panic!("No suitable `protoc` (>= 3.1.0) found in PATH"), 
    };
```

Indeed, the previous code (v0.14.0) appears to be broken. 
The TiKV team has already implemented updates, 
hence it would be beneficial to alter this crate's version to its most recent release, 
which is version 0.15.1.
This change will enable the successful building of the project on M1 Macs.

Please explain IN DETAIL what the changes are in this PR and why they are needed:

- Summarize your change (**mandatory**)
   changed the version of protobuf-build in `src/log-store/Cargo.toml` to 0.15.1.
  

## Checklist

- [x]  I have written the necessary rustdoc comments.
- [x]  I have added the necessary unit tests and integration tests.

### CODE REVIEW

File: src/log-store/Cargo.toml [hunk 1]
```diff
@@ -5,7 +5,7 @@ edition.workspace = true
 license.workspace = true
 
 [build-dependencies]
-protobuf-build = { version = "0.14", default-features = false, features = [
+protobuf-build = { version = "0.15.1", default-features = false, features = [
```
Comment (v0y4g3r) [inline src/log-store/Cargo.toml:8]: Does this also work?
```suggestion
protobuf-build = { version = "0.15", default-features = false, features = [
```

Comment (xie-zheng) [reply src/log-store/Cargo.toml:8]: Yes, this works too.

Comment (xie-zheng) [reply src/log-store/Cargo.toml:8]: The update of "protoc lookup" happened at 0.14.1, so 0.15 clearly works.

**Review by waynexia — APPROVED** [review]
LGTM

**Comment by waynexia** [comment]
Does this change the lock file `Cargo.lock`?

**Comment by xie-zheng** [comment]
> Does this change the lock file `Cargo.lock`?

I think the Cargo.lock will be changed when running `cargo build` or `cargo run`.

**Review by WenyXu — APPROVED** [review]

**Review by Nateiru — APPROVED** [review]
LGTM！

**Pull request merged**